rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -163,6 +172,55 @@ void registerBtcTransaction_whenLegacyBtcTransactionWithNegativeHeight_shouldNot
         assertEquals(expectedReceiverBalance, repository.getBalance(rskReceiver));
     }
 
+    @Test
+    void registerBtcTransaction_whenLegacyBtcTransactionFromAMultiSig_shouldRefundTheFunds() throws Exception {
+        // Arrange
+        BtcTransaction bitcoinTransaction = createMultiSigPegInTransaction(federationSupport.getActiveFederation().getAddress(), minimumPeginValue);
+        setupChainWithBtcTransaction(bitcoinTransaction);
+        bridgeSupport.save();
+
+        co.rsk.core.Coin expectedReceiverBalance = repository.getBalance(rskReceiver);
+        List<UTXO> expectedFederationUTXOs = List.copyOf(federationSupport.getActiveFederationBtcUTXOs());
+
+        // Act
+        bridgeSupport.registerBtcTransaction(rskTx, bitcoinTransaction.bitcoinSerialize(), btcBlockWithPmtHeight, pmtWithTransactions.bitcoinSerialize());
+        bridgeSupport.save();
+
+        // Assert
+        Set<PegoutsWaitingForConfirmations.Entry> pegoutEntries = bridgeStorageProvider.getPegoutsWaitingForConfirmations().getEntries();
+        int expectedPegoutsWaitingForConfirmations = 1;
+        assertEquals(expectedPegoutsWaitingForConfirmations, pegoutEntries.size());
+
+        Optional<PegoutsWaitingForConfirmations.Entry> pegOutWaitingForConfirmationOptional = pegoutEntries.stream().findFirst();
+        assertTrue(pegOutWaitingForConfirmationOptional.isPresent());
```

Comment:
Maybe we can do the findFirst or size strategy for both of them. For consistency.

Reply:
Not sure what you mean, I use both approaches to say: that it's the only one and it's exactly the one I am expecting

Reply:
I see, the size assertion indirectly ensures that there is exactly one pegout entry, making the second assertion somewhat redundant. Not a big deal, but you can simplify the code to have a single assert while maintaining clarity.

Reply:
Okay, I always wonder whether asserting an isPresent with an optional makes sense. The list of PegoutsWaitingForConfirmations.Entry may have a list of empty optionals

Reply:
Since you have checked that it has a size 1 it is guaranteed that the Optional is present.

Reply:
List.of(Optional.empty()).size() is equal to 1, that's why I check both